WAFL-Testbed/ctrl/main.py

```diff
@@ -21,14 +22,270 @@ class WaflAgent:
+        timeout: int = 10,
+    ):
         self.name = device_name
         self.ip = ip_address
         self.ctrl_port = ctrl_port
         self.status = "UNKNOWN"
         self.logger = logging.getLogger(f"WaflAgent-{device_name}")
         self.pid = None
         self.timeout = timeout
+        self.config = config
+
+        # Deploy configurations during initialization
+        self._deploy_configurations(experiment_parameters)
+
+    def _create_unified_config(self, experiment_parameters: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Create unified configuration for this agent.
+
+        Args:
+            experiment_parameters: Experiment parameters from ControlServer
+
+        Returns:
+            Dict containing unified configuration for this agent
+        """
+        unified_config = {
+            "agent_info": {
+                "device_name": self.name,
+                "ip_address": self.ip,
+            },
+            "experiment_info": {
+                "experiment_id": experiment_parameters.get("experiment_id"),
+                "experiment_name": self.config.get("EXPERIMENT_NAME", "exp"),
+            },
+            "infrastructure": {
+                "device_names": self.config["WAFL_DEVICE_NAMES"],
+                "device_ips": self.config["WAFL_DEVICE_IPS"],
+                "ctrl_port": self.config["WAFL_DEVICE_CTRL_PORT"],
+                "p2p_port": self.config["WAFL_DEVICE_P2P_PORT"],
+            },
+            "experiment_parameters": {
+                "epochs": experiment_parameters.get("epochs"),
+                "wafl_phase_params": experiment_parameters.get("wafl_phase_params", {}),
+            },
+            "runtime": {
+                "log_level": os.environ.get("LOG_LEVEL", "INFO"),
+            },
+        }
+
+        self.logger.debug(f"🔧 Created unified configuration for agent {self.name}")
+        return unified_config
+
+    def _deploy_configurations(self, experiment_parameters: Dict[str, Any]) -> bool:
+        """
+        Deploy all configuration files (contact pattern and agent config) to this agent via SSH.
+
+        Args:
+            experiment_parameters: Experiment parameters containing configuration data
+
+        Returns:
+            bool: True if all deployments successful, False otherwise
+        """
+        self.logger.info(f"📋 Deploying configurations to agent {self.name}")
+
+        try:
+            ssh_port = 22
+            username = self.config["USER"]
+            private_key_path = os.path.expanduser("~/.ssh/id_ed25519")
+
+            if not os.path.exists(private_key_path):
+                raise FileNotFoundError(f"🔑 SSH private key not found at {private_key_path}")
+
+            key = paramiko.Ed25519Key.from_private_key_file(private_key_path)
+            target_path = os.path.join(self.config["DEPLOYMENT_LOCATION"], "wafl")
+            config_dir = os.path.join(target_path, "config", "common")
```

Comment:
実行サーバーのディレクトリ構成に合ってなさそう
```suggestion
            target_path = os.path.join(self.config["DEPLOYMENT_LOCATION"], self.config.get("EXPERIMENT_NAME", "exp"))
            config_dir = os.path.join(target_path, "config")
```